### PR TITLE
prepare subprojects.settings.gradle for kotlin

### PR DESCRIPTION
**Pull request by soloturn**
make the files similar in logic, as well use syntax a little closer to kotlin, to then mirgrate and have a clearer, easier to review diff vew.

### CODE REVIEW

File: subsystems/subprojects.settings.gradle [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
-        def subprojectPath = ':' + subprojectName
+        def subprojectName = "subsystems:" + possibleSubprojectDir.name
+        logger.info("Including {} as a sub-project.", subprojectName)
+        include(subprojectName)
+        def subprojectPath = ":" + subprojectName
         def subproject = project(subprojectPath)
         subproject.projectDir = possibleSubprojectDir
     } else {
-        logger.info("Ignoring hidden folder '$possibleSubprojectDir'")
+        logger.info("Ignoring hidden folder {}.", possibleSubprojectDir)
```
Comment (BenjaminAmos) [inline subsystems/subprojects.settings.gradle]:
```suggestion
        logger.info("Ignoring hidden folder '{}'.", possibleSubprojectDir)
```

It helps sometimes to see quotation marks around file names to visualise any unexpected spaces in them.

File: subsystems/subprojects.settings.gradle [hunk 2]
```diff
@@ -2,15 +2,15 @@
...
 // This magically allows subdirs in this subproject to themselves become sub-subprojects in a proper tree structure
-new File(rootDir, 'subsystems').eachDir { possibleSubprojectDir ->
+new File(rootDir, "subsystems").eachDir { possibleSubprojectDir ->
     if (!possibleSubprojectDir.name.startsWith(".")) {
-        def subprojectName = 'subsystems:' + possibleSubprojectDir.name
-        logger.info("Including '$subprojectName' as a sub-project")
-        include subprojectName
-        def subprojectPath = ':' + subprojectName
+        def subprojectName = "subsystems:" + possibleSubprojectDir.name
+        logger.info("Including {} as a sub-project.", subprojectName)
```
Comment (BenjaminAmos) [inline subsystems/subprojects.settings.gradle]:
```suggestion
        logger.info("Including '{}' as a sub-project.", subprojectName)
```

File: libs/subprojects.settings.gradle [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
+    if (!buildFile.exists()) {
+        logger.warn("***** WARNING: Found a lib without a build.gradle, corrupt dir? NOT including {} *****", subprojectName)
+        return
+    }
+    if (!settingsFile.exists()) {
+        logger.warn("lib {} has build.gradle, but no settings.gradle? NOT including it." subprojectName)
+        return
     }
+    logger.info("lib {} has a build file so counting it complete and including it.", subprojectName)
+    include(subprojectName)
```
Comment (BenjaminAmos) [inline libs/subprojects.settings.gradle]: You've changed this from using `includeBuild` to `include`. Is this likely to have implications? An `includeBuild` is usually used for separate third-party projects, whilst `include` is typically used for projects within your own tree.

Comment (soloturn) [reply libs/subprojects.settings.gradle]: fixed, @BenjaminAmos,

File: libs/subprojects.settings.gradle [hunk 4]
```diff
@@ -1,21 +1,20 @@
...
+    if (!buildFile.exists()) {
+        logger.warn("***** WARNING: Found a lib without a build.gradle, corrupt dir? NOT including {} *****", subprojectName)
+        return
+    }
+    if (!settingsFile.exists()) {
+        logger.warn("lib {} has build.gradle, but no settings.gradle? NOT including it." subprojectName)
+        return
     }
+    logger.info("lib {} has a build file so counting it complete and including it.", subprojectName)
+    includeBuild(subprojectName)
```
Comment (BenjaminAmos) [inline libs/subprojects.settings.gradle]:
```suggestion
// This magically allows subdirs in this subproject to themselves become sub-subprojects in a proper tree structure
new File(rootDir, 'libs').eachDir { possibleSubprojectDir ->
    def subprojectName = ':libs:' + possibleSubprojectDir.name
    File buildFile = new File(possibleSubprojectDir, "build.gradle")
    File settingsFile = new File(possibleSubprojectDir, "settings.gradle")
    if (!buildFile.exists()) {
        logger.warn("***** WARNING: Found a lib without a build.gradle, corrupt dir? NOT including {} *****", subprojectName)
        return
    }
    if (!settingsFile.exists()) {
        logger.warn("lib {} has build.gradle, but no settings.gradle? NOT including it." subprojectName)
        return
    }
    logger.info("lib {} has a build file so counting it complete and including it.", subprojectName)
    includeBuild(possibleSubprojectDir)
```

`includeBuild` takes-in a file, not a file path. These changes do not actually work without it.

You can test this by doing:
 - `./groovyw lib get TeraNUI`
 - `./gradlew :TeraNUI:build`

**Review by BenjaminAmos — DISMISSED** [review]

File: libs/subprojects.settings.gradle (hunk 4 above)
Comment (soloturn) [reply libs/subprojects.settings.gradle]: fixed. added this as a comment so one sees this when looking at the source code already.

**Review by BenjaminAmos — APPROVED** [review]

**Pull request merged**